### ISSUE TITLE
Loss value is Nan when running curriculum training.

### ISSUE THREAD

**Issue by haithanhp** [jimeiyang/deepRotator#1]
Hi Jimeiyang,

Thank for your great work. I downloaded and ran the code and I got Nan value for curriculum training. I start with numsteps of 2 and use base_model_lr1e-4_iter0400.mat. When I debug, I saw that forward function output Nan values. Does it happen on your side when training that?

**Comment by jimeiyang** [comment]
Have you trained the model using "train_chair_rotator_base.m" before running the curriculum training?

**Comment by haithanhp** [comment]
Hi. I have trained model with base (500 iterations). I used your generator function python to re-generate rnn_t2/4/8/16 and now the loss value is not Nan longer.  By the way, training progress is pretty slow, Can I convert into C++ to utilize multiple gpus? Is there any notice to convert it? Thanks.

**Comment by jimeiyang** [comment]
Glad to know that. Matcaffe training is indeed slow. It will be great if you'd like to convert to C++. Let me know when you have it. You may take a look at the torch implementation of deepRotator here: https://github.com/xcyan/nips16_PTN as reference.

**Comment by haithanhp** [comment]
Thank you for providing me the precious project. Btw, in your code when calculate euclidean gradient, I notice you did not divide to size of array (**grad = single(pred-output);** ). Is there difference between divide and not divide which may impact the final result? I want to know the detail when convert them. Thanks.

**Comment by jimeiyang** [comment]
Should not be different, but the learning rate needs to be adjusted accordingly.

**Comment by xcyan** [comment]
@HaiPhan1991 Please also check the TensorFlow Implementation that pre-trains the deepRotator (NIPS'15): https://github.com/tensorflow/models/blob/master/research/ptn/pretrain_rotator.py

**Comment by haithanhp** [comment]
Awesome! Thanks for your kind support. I will check it for my research.
